api/addons/v1beta1/conversion.go

```diff
@@ -17,9 +17,12 @@ limitations under the License.
 package v1beta1
```

Comment:
There are a few more cases where we have to set Conditions to nil
      * v1alpha3.Cluster ConvertTo + TODO for ControlPlaneInitializedCondition
      * v1alpha3.Machine,MachineSet,MachineDeployment,MachineHealthCheck ConvertTo